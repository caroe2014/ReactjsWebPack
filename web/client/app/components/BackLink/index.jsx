// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { Flex } from 'rebass';
import styled from 'styled-components';
import { Trans } from 'react-i18next';
import i18n from 'web/client/i18n';
const BackArrow = styled(Flex)`
  & > i {
    -webkit-text-stroke: 1px #f0f0f3;
  }
  color: ${props => {
    if (props.theme.colors.reverseTextColor) return `${props.theme.colors.reverseTextColor}`;
    return 'transparent';
  }};
${props => props.theme.mediaBreakpoints.desktop`
    display: ${props => {
    if (((props.section === 'site' || props.section === 'loyalty') && !props['hide-arrow']) ||
     (props.section === 'home' && props['hide-arrow']) || props['multi-pass']) return `none`;
    return 'block';
  }};
  color: ${props => {
    if (props.theme.colors.buttonControlColor) return `${props.theme.colors.buttonControlColor}`;
    return 'transparent';
  }};
`
}
${props => props.theme.mediaBreakpoints.tablet`
  display: ${props => {
    if ((props.section === 'home' && !props['hide-arrow'])) return `none`;
    return 'block';
  }};  
  padding-left: 8px; padding-right: 8px;`}
${props => props.theme.mediaBreakpoints.mobile`
  display: ${props => {
    if ((props.section === 'home' && !props['hide-arrow'])) return `none`;
    return 'block';
  }};  
  padding-left: 8px; padding-right: 8px;`
}
`;
const BackArrowText = styled(Flex)`
color: ${props => {
    if (props.theme.colors.buttonControlColor) return `${props.theme.colors.buttonControlColor}`;
    return 'transparent';
  }};
`;
class BackLink extends Component {
  constructor (props) {
    super(props);
    this.state = {
      linkText: '',
      previousPath: '',
      showBackLink: true
    };
    this.handleChange = this.handleChange.bind(this);
    this.onNavLinkClick = this.onNavLinkClick.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
  }
  componentDidMount () {
    this.handleChange(this.props);
  }
  componentWillReceiveProps (nextProps) {
    this.handleChange(nextProps);
  }
  handleChange (nextProps) {
    const { match, menu, section, lastCartLocation, location, multiPassEnabled,
      deliveryLocation, tipCapture, isSmsEnabled, nameCapture, sites, digitalMenuId, conceptList } = nextProps;

    switch (section) {
      case 'item':
        this.setState({
          linkText: <Trans i18nKey='CONTEXT_BACK_TO_ITEMS'/>,
          previousPath: `/menu/${match.params.siteId}/${match.params.displayProfileId}/${match.params.conceptId}/${match.params.categoryId}` // eslint-disable-line max-len
        });
        break;
      case 'menu':
        let category = match.params.categoryId && menu.current.categories
          ? menu.current.categories.find(category => category.id === match.params.categoryId)
          : {};
        let linkText = category.name
          ? <Trans i18nKey='CONTEXT_BACK_TO_CATEGORIES'/>
          : <Trans i18nKey='CONTEXT_BACK_TO_CONCEPTS'/>;
        let path = category.name
          ? `/menu/${match.params.siteId}/${match.params.displayProfileId}/${match.params.conceptId}`
          : `/concepts/${match.params.siteId}/${match.params.displayProfileId}`;
        let showBackLink = digitalMenuId
          ? category.name ? true : (conceptList && conceptList.length > 1)
          : true;
        this.setState({ linkText: linkText, previousPath: path, showBackLink });
        break;
      case 'site':
        const { skipSitesPage } = sites;
        if (skipSitesPage) {
          this.setState({
            linkText: i18n.t('CONTEXT_BACK_TO_CHANGE_TIME'),
            previousPath: '/',
            showBackLink: !digitalMenuId
          });
          break;
        } else {
          this.setState({
            linkText: <Trans i18nKey={'CONTEXT_BACK_CHANGE_LOCATION'}/>, // eslint-disable-line max-len
            previousPath: `/`,
            showBackLink: !digitalMenuId
          });

          break;
        }
      case 'home':
        this.setState({ linkText: i18n.t('CONTEXT_BACK_TO_CHANGE_TIME'), previousPath: `/`, showBackLink: !digitalMenuId }); // eslint-disable-line max-len
        break;
      case 'payment':
        if (location.pathname && location.pathname === '/paymentComponent') {
          this.setState({ linkText: <Trans i18nKey={'CONTEXT_BACK_TO'}/>,
            previousPath: '/payment'
          });
        } else if (deliveryLocation && !multiPassEnabled) {
          this.setState({ linkText: <Trans i18nKey={'CONTEXT_BACK_TO_DELIVERY'}/>,
            previousPath: '/deliveryLocation'
          });
        } else if (isSmsEnabled) {
          this.setState({ linkText: <Trans i18nKey={'CONTEXT_BACK_TO_SMS'}/>,
            previousPath: '/smsNotification'
          });
        } else if (nameCapture && !multiPassEnabled) {
          this.setState({ linkText: <Trans i18nKey={'CONTEXT_BACK_TO_NAME_CAPTURE'}/>,
            previousPath: '/nameCapture'
          });
        } else if (tipCapture) {
          this.setState({ linkText: <Trans i18nKey={'CONTEXT_BACK_TO_TIP'}/>,
            previousPath: '/tip'
          });
        } else {
          this.setState({ linkText: <Trans i18nKey={'CONTEXT_BACK_TO'}/>,
            previousPath: lastCartLocation
          });
        }
        break;
      case 'delivery':
        if (isSmsEnabled) {
          this.setState({ linkText: <Trans i18nKey={'CONTEXT_BACK_TO_SMS'}/>,
            previousPath: '/smsNotification'
          });
        } else if (nameCapture) {
          this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO_NAME_CAPTURE'/>,
            previousPath: '/nameCapture'
          });
          break;
        } else if (tipCapture) {
          this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO_TIP'/>,
            previousPath: ((location.pathname && location.pathname === '/deliveryLocation') ? '/tip' : '/deliveryLocation') // eslint-disable-line max-len
          });
          break;
        } else {
          this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO'/>,
            previousPath: ((location.pathname && location.pathname === '/deliveryLocation') ? lastCartLocation : '/deliveryLocation') // eslint-disable-line max-len
          });
          break;
        }
        break;
      case 'namecapture':
        if (tipCapture) {
          this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO_TIP'/>,
            previousPath: ((location.pathname && location.pathname === '/nameCapture') ? '/tip' : '/nameCapture') // eslint-disable-line max-len
          });
          break;
        } else {
          this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO'/>,
            previousPath: ((location.pathname && location.pathname === '/nameCapture') ? lastCartLocation : '/nameCapture') // eslint-disable-line max-len
          });
          break;
        }

      case 'tip':
        this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO'/>,
          previousPath: ((location.pathname && location.pathname === '/tip') ? lastCartLocation : '/tip') // eslint-disable-line max-len
        });
        break;
      case 'smsnotification':
        if (nameCapture && !multiPassEnabled) {
          this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO_NAME_CAPTURE'/>,
            previousPath: '/nameCapture'
          });
          break;
        } else if (tipCapture) {
          this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO_TIP'/>,
            previousPath: '/tip'
          });
          break;
        } else {
          this.setState({ linkText: <Trans i18nKey='CONTEXT_BACK_TO'/>,
            previousPath: lastCartLocation
          });
          break;
        }
    }
  }

  onNavLinkClick () {
    if (this.state.linkText === i18n.t('CONTEXT_BACK_TO_CHANGE_TIME')) {
      this.props.setHideScheduleTime(false);
    }
  }

  hideBackLinkOnMultiPass () {
    const { match, menu, section, location, multiPassEnabled } = this.props;
    let category = match.params.categoryId && menu.current.categories
      ? menu.current.categories.find(category => category.id === match.params.categoryId)
      : {};
    return multiPassEnabled && (location.pathname.includes('concepts') ||
    (section === 'menu' && (!category.name && category.name !== '')));
  }

  handleKeyDown (e) {
    if (e.which === 32 || e.which === 13) {
      this.onNavLinkClick();
    }
  }

  render () {
    const { showText, section, hideScheduleTime, isScheduleOrderEnabled, siteFetch } = this.props;
    const { linkText, previousPath, showBackLink } = this.state;
    const isHideBackLinkOnMultiPass = this.hideBackLinkOnMultiPass();
    return (
      <React.Fragment>
        {showBackLink && !siteFetch &&
          <NavLink style={{ display: 'flex', outline: 'none', boxShadow: 'none' }}
            aria-label={i18n.t('NAVIGATE_PREV_PAGE')} tabIndex={showText && linkText ? -1 : 0}
            onKeyDown={this.handleKeyDown}
            className='back-arrow-link' to={previousPath} onClick={this.onNavLinkClick}>
            <BackArrow hide-arrow={hideScheduleTime}
              multi-pass={isHideBackLinkOnMultiPass}
              schedule-order={isScheduleOrderEnabled} section={section}>
              <i className='fa fa-arrow-left' />
            </BackArrow>
            {showText && !(isHideBackLinkOnMultiPass) &&
            <BackArrowText role='button' aria-label={i18n.t('NAVIGATE_PREV_PAGE')}
              tabIndex={linkText ? 0 : -1}>&nbsp;{linkText}</BackArrowText>}
          </NavLink>
        }
      </React.Fragment>
    );
  }
}
export default BackLink;
