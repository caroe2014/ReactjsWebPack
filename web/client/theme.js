// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
