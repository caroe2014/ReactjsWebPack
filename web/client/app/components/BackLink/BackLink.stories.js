// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { Provider } from 'rebass';
import theme from 'web/client/theme.js';
import BackLink from 'web/client/app/components/BackLink';

const props = {
  match: {
    params: {}
  }
};

storiesOf('BackLink', module)
  .addDecorator(withKnobs)
  .add('BackLink', () => {
    return (
      <div>
        <Provider theme={theme}>
          <BackLink section='site' {...props}/>
        </Provider>
      </div>
    );
  });
