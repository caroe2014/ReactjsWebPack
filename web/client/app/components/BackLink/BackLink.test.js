// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {
  shallow,
  mount,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import { BrowserRouter as Router } from 'react-router-dom';
import { t } from 'i18next';

import BackLink from '.';

configure({
  adapter: new Adapter()
});

jest.mock('i18next');
t.mockImplementation((key) => {
  return key;
});

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('BackLink', () => {
  var mockMatch = {
    params: {}
  };
  const sites = {};
  it('does a snapshot test', () => {
    const wrapper = mount(<Router><ThemeProvider theme={theme}><BackLink showText section='site' match={mockMatch} sites={sites}/></ThemeProvider></Router>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error when section is site', () => {
    expect(shallow(<BackLink showText section='site' match={mockMatch} sites={sites}/>).exists()).toBe(true);
  });

  it('should render with updated themes', () => {
    theme.colors.reverseTextColor = '#000000';
    theme.colors.buttonControlColor = '#000000';
    expect(mount(<Router><ThemeProvider theme={theme}><BackLink showText section='site' match={mockMatch} sites={sites}/></ThemeProvider></Router>).exists()).toBe(true);
  });

  it('should render without throwing an error when section is Item', () => {
    expect(shallow(<BackLink showText section='item' match={mockMatch} />).exists()).toBe(true);
  });

  describe('when section is menu', () => {

    it('should render without throwing an error when section is menu', () => {
      expect(shallow(<BackLink showText section='menu' match={mockMatch} />).exists()).toBe(true);
    });

    it('traverse and get category when menu is provided', () => {
      var mockLocalMatch = {
        params: {
          categoryId: 1
        }
      };
      var menu = {
        current: {
          categories: [
            {
              id: 1,
              name: 'test'
            }
          ]
        }
      };
      expect(shallow(<BackLink showText section='menu' match={mockLocalMatch} menu={menu} />).exists()).toBe(true);
    });
  });

  describe('when section is payment', () => {
    it('should render without throwing an error when section is payment', () => {
      var location = {
        pathname: '/paymentComponent'
      };
      const pathWrapper = shallow(<BackLink showText section='payment' location={location} match={mockMatch} />);
      expect(pathWrapper.exists()).toBe(true);
      expect(pathWrapper.state().previousPath).toBe('/payment');
    });

    it('should redirect to Sms notification if Sms is enabled', () => {
      const redirectWrapper = shallow(<BackLink showText section='payment' location isSmsEnabled match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/smsNotification');
    });

    it('should redirect to deliveryLocation if delivery is enabled', () => {
      const redirectWrapper = shallow(<BackLink showText section='payment' location deliveryLocation match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/deliveryLocation');
    });

    it('should redirect to nameCapture if nameCapture is enabled', () => {
      const redirectWrapper = shallow(<BackLink showText section='payment' location nameCapture match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/nameCapture');
    });

    it('should redirect to tipCapture if tipCapture is enabled', () => {
      const redirectWrapper = shallow(<BackLink showText section='payment' location tipCapture match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/tip');
    });

    it('should redirect to last cart location if none of the above are enabled', () => {
      const redirectWrapper = shallow(<BackLink showText section='payment' lastCartLocation='test' location match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('test');
    });
  });

  describe('when section is delivery', () => {
    it('should render without throwing an error when section is delivery', () => {
      var location = {
        pathname: '/deliveryLocation'
      };
      expect(shallow(<BackLink showText section='delivery' location={location} match={mockMatch} />).exists()).toBe(true);
    });

    it('should redirect to  deliveryLocation when tip is not enabled and but pathname is not deliveryLocation', () => {
      var location = {
        pathname: '/test'
      };
      const redirectWrapper = shallow(<BackLink showText section='delivery' location={location} match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/deliveryLocation');
    });

    it('should redirect to tipCapture if tipCapture is enabled', () => {
      var location = {
        pathname: '/deliveryLocation'
      };
      const redirectWrapper = shallow(<BackLink showText section='delivery' tipCapture location={location} match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/tip');
    });

    it('should redirect to deliverylocation if  if tipCapture is enabled but pathname is not deliveryLocation', () => {
      var location = {
        pathname: '/test'
      };
      const redirectWrapper = shallow(<BackLink showText section='delivery' tipCapture location={location} match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/deliveryLocation');
    });
  });

  describe('when section is namecapture', () => {
    it('should render without throwing an error when section is namecapture', () => {
      var location = {
        pathname: '/nameCapture'
      };
      expect(shallow(<BackLink showText section='namecapture' location={location} match={mockMatch} />).exists()).toBe(true);
    });

    it('should redirect to nameCapture when pathname is not nameCapture', () => {
      var location = {
        pathname: '/test'
      };
      const redirectWrapper = shallow(<BackLink showText section='namecapture' location={location} match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/nameCapture');
    });

    it('should redirect to tip when pathname is nameCapture', () => {
      var location = {
        pathname: '/nameCapture'
      };
      const redirectWrapper = shallow(<BackLink showText section='namecapture' tipCapture location={location} match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/tip');
    });

    it('should redirect to nameCapture when pathname is not nameCapture but tip is enabled', () => {
      var location = {
        pathname: '/test'
      };
      const redirectWrapper = shallow(<BackLink showText section='namecapture' tipCapture location={location} match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/nameCapture');
    });
  });

  describe('when section is tip', () => {
    it('should render without throwing an error when section tip', () => {
      var location = {
        pathname: '/tip'
      };
      expect(shallow(<BackLink showText section='tip' location={location} match={mockMatch} />).exists()).toBe(true);
    });
    it('should redirect to tip when path name is not tip', () => {
      var location = {
        pathname: '/test'
      };
      const redirectWrapper = shallow(<BackLink showText section='tip' location={location} match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/tip');
    });
  });

  describe('when section is smsnotification', () => {
    it('should render without throwing an error when section smsnotification', () => {
      var mockMatch = {
        params: {}
      };
      expect(shallow(<BackLink showText section='smsnotification' match={mockMatch} />).exists()).toBe(true);
    });

    it('should redirect to deliveryLocation if enabled', () => {
      var mockMatch = {
        params: {}
      };
      const redirectWrapper = shallow(<BackLink showText location section='payment' deliveryLocation match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/deliveryLocation');
    });

    it('should redirect to nameCapture if enabled', () => {
      var mockMatch = {
        params: {}
      };
      const redirectWrapper = shallow(<BackLink showText section='smsnotification' nameCapture match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/nameCapture');
    });

    it('should redirect to tipCapture if enabled', () => {
      var mockMatch = {
        params: {}
      };
      const redirectWrapper = shallow(<BackLink showText section='smsnotification' tipCapture match={mockMatch} />);
      expect(redirectWrapper.state().previousPath).toBe('/tip');
    });

  });
});
