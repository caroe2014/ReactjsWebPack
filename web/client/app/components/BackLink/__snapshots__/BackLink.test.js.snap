// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackLink does a snapshot test 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "colors": Object {
            "addOnTab": "#347CFF",
            "addToCart": "#347CFF",
            "addonBg": "#EEEEEE",
            "backgroundGrey": "#E6E6E6",
            "black": "#000000",
            "close": "red",
            "contextBackground": "#f0f0f3",
            "darkGrey": "#464646",
            "disableBackground": "#E4E4E4",
            "disableButton": "#D1D1D1",
            "error": "#cc0000",
            "footer": "#48a942",
            "glutenFree": "#8d7356",
            "healthy": "#359919",
            "hint": "#a6a6a6",
            "inputBottomBorder": "lightgray",
            "itemBorder": "lightgrey",
            "itemTile": "#3C474C0",
            "light": "white",
            "loader": "#ababab",
            "loginBackground": "#303030",
            "loginBoxColor": "#434149",
            "loginField": "#AAA7AC",
            "loginFormBackground": "#484848",
            "navigationMenu": "#48a942",
            "navigationMenuHover": "cornflowerblue",
            "new": "#ff6000",
            "open": "#48a942",
            "paymentSuccessTitle": "#505050",
            "placeHolder": "#909090",
            "primary": "#347CFF",
            "primaryButton": "#347CFF",
            "primaryText": "#383838",
            "secondaryText": "#a6a6a6",
            "spicy": "#ed1c24",
            "text": "black",
            "textGrey": "#505050",
            "tile": "#505050",
            "validationError": "#cc0000",
            "vegan": "#0d4a22",
            "vegetarian": "#0d4a22",
          },
          "fontFamily": "Roboto",
          "fontSize": Object {
            "lg": "1.8rem",
            "md": "1.3rem",
            "nm": "1rem",
            "sm": "0.75rem",
            "xl": "2.4rem",
          },
          "fontSizeFromProps": [Function],
          "mediaBreakpoints": Object {
            "desktop": [Function],
            "mobile": [Function],
            "tablet": [Function],
          },
          "nav": Object {
            "height": "70px",
          },
          "receiptFontFamily": "Roboto",
          "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
          "spacing": Object {
            "lg": "4em",
            "md": "2em",
            "nm": "1em",
            "none": "0em",
            "sm": "0.5em",
            "xl": "8em",
          },
        }
      }
    >
      <BackLink
        match={
          Object {
            "params": Object {},
          }
        }
        section="site"
        showText={true}
        sites={Object {}}
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          aria-label="NAVIGATE_PREV_PAGE"
          className="back-arrow-link"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "boxShadow": "none",
              "display": "flex",
              "outline": "none",
            }
          }
          tabIndex={-1}
          to="/"
        >
          <Route
            path="\\\\/"
          >
            <Link
              aria-current="page"
              aria-label="NAVIGATE_PREV_PAGE"
              className="back-arrow-link active"
              onClick={[Function]}
              onKeyDown={[Function]}
              replace={false}
              style={
                Object {
                  "boxShadow": "none",
                  "display": "flex",
                  "outline": "none",
                }
              }
              tabIndex={-1}
              to="/"
            >
              <a
                aria-current="page"
                aria-label="NAVIGATE_PREV_PAGE"
                className="back-arrow-link active"
                href="/"
                onClick={[Function]}
                onKeyDown={[Function]}
                style={
                  Object {
                    "boxShadow": "none",
                    "display": "flex",
                    "outline": "none",
                  }
                }
                tabIndex={-1}
              >
                <Styled(Flex)
                  section="site"
                >
                  <Flex
                    className="sc-jqCOkK mBDlV"
                    section="site"
                  >
                    <Box
                      className="sc-jqCOkK mBDlV sc-bwzfXH hKiLMS"
                      section="site"
                    >
                      <div
                        className="sc-jqCOkK mBDlV sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                      >
                        <i
                          className="fa fa-arrow-left"
                        />
                      </div>
                    </Box>
                  </Flex>
                </Styled(Flex)>
                <Styled(Flex)
                  aria-label="NAVIGATE_PREV_PAGE"
                  role="button"
                  tabIndex={0}
                >
                  <Flex
                    aria-label="NAVIGATE_PREV_PAGE"
                    className="sc-uJMKN bzxWwV"
                    role="button"
                    tabIndex={0}
                  >
                    <Box
                      aria-label="NAVIGATE_PREV_PAGE"
                      className="sc-uJMKN bzxWwV sc-bwzfXH hKiLMS"
                      role="button"
                      tabIndex={0}
                    >
                      <div
                        aria-label="NAVIGATE_PREV_PAGE"
                        className="sc-uJMKN bzxWwV sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                        role="button"
                        tabIndex={0}
                      >
                        Â 
                        <div
                          i18nKey="CONTEXT_BACK_CHANGE_LOCATION"
                        />
                      </div>
                    </Box>
                  </Flex>
                </Styled(Flex)>
              </a>
            </Link>
          </Route>
        </NavLink>
      </BackLink>
    </ThemeProvider>
  </Router>
</BrowserRouter>
`;
